File: packages/ios/Sources/Components/FloorSelector/FloorSelector.figma.swift
import SwiftUI
import Figma

struct KozmosFloorSelectorConnect: FigmaConnect {
    let component = KozmosFloorSelector.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1340-6786"

    @FigmaString("Selected Floor Text")
    var selectedFloor: String = "L1"

    @FigmaEnum(
        "Variant",
        mapping: [
            "VerticalList": KozmosFloorSelectorVariant.verticalList,
            "HorizontalList": KozmosFloorSelectorVariant.horizontalList,
            "CompactStepper": KozmosFloorSelectorVariant.compactStepper
        ]
    )
    var floorSelectorVariant: KozmosFloorSelectorVariant = .verticalList

    // floors carries the canonical level IDs and their ordering, which the
    // Figma set shows as three examples but cannot express.
    var body: some View {
        KozmosFloorSelector(
            floors: ["L2", "L1", "G"],
            selectedFloor: .constant(self.selectedFloor),
            variant: self.floorSelectorVariant
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/FloorSelector/FloorSelector.figma.kt
package com.kozmos.components.floorselector

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1340-6786")
class KozmosFloorSelectorConnect {
    @FigmaProperty(FigmaType.Text, "Selected Floor Text")
    val selectedFloor: String = "L1"

    // Named floorSelectorVariant because Code Connect reserves `variant`.
    @FigmaProperty(FigmaType.Enum, "Variant")
    val floorSelectorVariant: KozmosFloorSelectorVariant = Figma.mapping(
        "VerticalList" to KozmosFloorSelectorVariant.VerticalList,
        "HorizontalList" to KozmosFloorSelectorVariant.HorizontalList,
        "CompactStepper" to KozmosFloorSelectorVariant.CompactStepper
    )

    @Composable
    fun ComponentExample() {
        KozmosFloorSelector(
            floors = listOf("L2", "L1", "G"),
            selectedFloor = selectedFloor,
            onFloorSelect = {},
            variant = floorSelectorVariant
        )
    }
}
